packages/layout/src/steps/resolvePagination.js

```diff
@@ -194,13 +196,44 @@ const splitPage = (page, pageNumber, fontStore) => {
   const height = R.path(['style', 'height'], page);
   const dynamicPage = resolveDynamicPage({ pageNumber }, page, fontStore);
 
-  const [currentChilds, nextChilds] = splitNodes(
+  const relayout = node => relayoutPage(node, fontStore);
+
+  let [currentChilds, nextChilds] = splitNodes(
     wrapArea,
     contentArea,
     dynamicPage.children,
   );
 
-  const relayout = node => relayoutPage(node, fontStore);
+  const getFootNotesView = footNotes =>
+    R.compose(
+      relayout,
+      assingChildren([mapFootNotesToView(footNotes)]),
+      R.assocPath(['box', 'height'], height),
+    )(page).children[0];
+
+  let pageFootNotes = getFootNotes({ children: currentChilds });
+  let footNotesView = getFootNotesView(pageFootNotes);
+
+  if (pageFootNotes.length > 0) {
+    [currentChilds, nextChilds] = splitNodes(
+      wrapArea - 25 - getHeight(footNotesView),
```

Comment:
Is `25` some arbitrary value? How is it being derieved?

Reply:
I've added this as a workaround for a rare text overlapping issue. I've removed it for now. need to find exactly how it happens
<img width="495" alt="Screenshot 2022-06-19 at 22 25 12" src="https://user-images.githubusercontent.com/43986725/174491939-5ee67c6d-664a-47b1-bb84-076d950c2119.png">

Reply:
fixed in new revision